File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
    maxCandy := 0
    for _, v := range candies {
        maxCandy = max(maxCandy, v)
    }
    
    ans := make([]bool, len(candies))
    for i, v := range candies {
        if v + extraCandies >= maxCandy {
            ans[i] = true
        }
    }

    return ans
}

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.py
class Solution(object):
    def kidsWithCandies(self, candies, extraCandies):
        """
        :type candies: List[int]
        :type extraCandies: int
        :rtype: List[bool]
        """
        greatestCandy = max(candies)
        res = []
        for candy in candies:
            if candy + extraCandies >= greatestCandy:
                res.append(True)
            else:
                res.append(False)
        
        return res